app/models/concerns/essi/dynamic_solr_document.rb

```diff
@@ -17,13 +17,21 @@ def load_allinson_flex
         # profile directly.
         profile = AllinsonFlex::Profile.current_version
         unless profile.blank?
-          profile.properties.each do |prop|
-            attribute(
-              prop.name,
-              # if the property is singular, make it so
-              prop.cardinality_maximum == 1 ? Hyrax::SolrDocument::Metadata::Solr::String : Hyrax::SolrDocument::Metadata::Solr::Array,
-              solr_name(prop.name.to_s)
-            )
+          Rails.logger.debug { "Loading AllinsonFlex DynamicSolrDocument #{ profile.profile_version } with #{profile.properties.count} properties." }
+
+          # Loading attributes is done by defining a method for each property. This only needs to be done once or when the profile changes.
```

Comment:
@dlpierce I know this just got merged, but once you feel like its battle tested, I think this would be something that Allinsonflex should just have by default. do you agree?

Reply:
Yes, certainly. There are a few enhancements we've come up with that we are planning to move into the gem.